scene/resources/shader.h

```diff
@@ -87,15 +87,32 @@ class Shader : public Resource {
 
 	virtual bool is_text_shader() const;
 
-	_FORCE_INLINE_ StringName remap_uniform(const StringName &p_uniform) const {
+	// Finds the shader uniform name for the given property name, which should start with "shader_uniform/".
+	_FORCE_INLINE_ StringName remap_uniform(const StringName &p_property) const {
 		if (params_cache_dirty) {
 			get_shader_uniform_list(nullptr);
 		}
 
-		const HashMap<StringName, StringName>::Iterator E = params_cache.find(p_uniform);
-		if (E) {
-			return E->value;
+		String n = p_property;
+
+		// Backwards compatibility with old shader parameter names.
+		if (n.begins_with("param/")) {
+			WARN_PRINT("Using legacy shader uniform name '" + n + "'. Use 'shader_param/...' instead.");
+			n = n.replace_first("param/", "shader_uniform/");
+		}
+		if (n.begins_with("shader_param/")) {
+			WARN_PRINT("Using legacy shader uniform name '" + n + "'. Use 'shader_param/...' instead.");
```

Comment:
```suggestion
			WARN_PRINT("Using legacy shader uniform name '" + n + "'. Use 'shader_uniform/...' instead.");
			n = n.replace_first("param/", "shader_uniform/");
		}
		if (n.begins_with("shader_param/")) {
			WARN_PRINT("Using legacy shader uniform name '" + n + "'. Use 'shader_uniform/...' instead.");
```
But I'm not sure it's worth a warning, this is not something that users *used* themselves, it's just an internal implementation detail for which we handle backward compat.

Reply:
Hmm, I was thinking some users use `set("shader_param/foo", 123)` in code. I guess a better alternative to either would be warn users when setting that way instead of `set_shader_param`.